src/main/java/com/minecolonies/coremod/commands/colonycommands/MakeNotAutoDeletable.java

```diff
@@ -0,0 +1,92 @@
+package com.minecolonies.coremod.commands.colonycommands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.commands.AbstractSingleCommand;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class MakeNotAutoDeletable extends AbstractSingleCommand
+{
+    public static final  String DESC                       = "undeletable";
+    private static final String NO_COLONY_FOUND_MESSAGE_ID = "Colony with ID %d not found.";
+    private static final String MARKED                     = "Marking succesful!";
+    private static final String NOT_ENOUGH_ARGUMENTS       = "You must have 2 Arguments: <ColonyId> <true|false> ";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public MakeNotAutoDeletable(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <true|false>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        if (args.length < 2)
+        {
+            sender.addChatMessage(new TextComponentString(NOT_ENOUGH_ARGUMENTS));
+            return;
+        }
+
+        sender.addChatMessage(new TextComponentString(Arrays.toString(args)));
+
+        int colonyId;
+        colonyId = Integer.parseInt(args[0]);
+        Colony colony = ColonyManager.getColony(colonyId);
+
+        boolean canBeDeleted;
+        canBeDeleted = Boolean.parseBoolean(args[1]);
+
+        final Entity senderEntity = sender.getCommandSenderEntity();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "senderEntity". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "senderEntity" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1481)